### ISSUE TITLE
Accessibility issues with NVDA

### ISSUE THREAD

**Issue by MowWow** [ng-select/ng-select#699]
Hi,

I tried to validate accessibility of the ng-select using NVDA on Google Chrome. I noticed some issues, and was wondering if it's possible for you to fix them.

I ran the validation on this page : https://ng-select.github.io/ng-select#/multiselect-checkbox

First issue : i have already chosen two options, Karyn Wright and Mendoza Ruiz, but when the focus enters the input, the vocal reader says "empty". Is it possible to make it read the actual selected options?

Second issue : when i navigate the options using the keybord, the vocal reader always says "Not selected" for all the options. Is says " Karyn Wright not selected", "Elsa Bradley not selected", even when the option is selected, the reader says Not selected. Is is possible to fix this too?

Thanks a lot

**Comment by abhishekgupta1991** [comment]
Hi, 
One more issue we face with nvda. If we pass required attribute, NVDA is not mention required during voice.

**Comment by abhishekgupta85** [comment]
ssue with NVDA is If we pass required attribute, NVDA is not mention required during screen reader.

**Comment by kittykatattack** [comment]
Just confirming we are having the same problem with JAWS. My QA team is not happy about this and I'm looking for a work-around. Does anyone have any suggestions?

**Comment by filipeovercom** [comment]
Hi, I have the same problem! Does anyone have a work-around?

**Comment by brunano21** [comment]
Along with other accessibility issues, I see that NVDA (and I guess JAWS too) does not read widget's value. This is due to the fact that the inner input's `value` is always empty.
@varnastadeus @anjmao would you be willing to accept any PR fixing those issues or to work on that?

**Comment by brunano21** [comment]
I have some work for better accessibility support. https://github.com/ng-select/ng-select/pull/1026

**Comment by agenaille** [comment]
I have just applied a workaround in our codebase which sets the aria-label on the <input /> to read the field's label and the selected value(s). It would be nice to get an official fix for this though. I do realize the complexity when the user is supplying a custom label template, though.

**Comment by sarnecki-cic** [comment]
> I have just applied a workaround in our codebase which sets the aria-label on the to read the field's label and the selected value(s). It would be nice to get an official fix for this though. I do realize the complexity when the user is supplying a custom label template, though.

@agenaille Can I ask you for more thorough explanation? How did you do that?

**Comment by dominiknovicky** [comment]
@agenaille Inject ElementRef and in the ngAfterViewInit set the attribute of the ng-select input element.

constructor(private readonly elementRef: ElementRef) {}
ngAfterViewInit(): void {
  const input = this.elementRef.nativeElement.children[0].querySelector('input');
  input.setAttribute('aria-required', 'true');
}

Hopefully, it helps.

**Comment by pavankjadda** [comment]
Closed part of latest release. Check and let mw know
